File: backend/app/schemas/resume.py
from pydantic import BaseModel


class ResumeAnalysis(BaseModel):
    score: int
    match_score: int

    ats_level: str
    summary: str

    formatting_score: int
    keyword_score: int
    experience_score: int
    skills_score: int
    education_score: int
    readability_score: int

    strengths: list[str]
    weaknesses: list[str]

    matching_skills: list[str]
    missing_skills: list[str]

    suggestions: list[str]


class ResumeUploadResponse(BaseModel):
    filename: str
    pages: int
    analysis: ResumeAnalysis

File: frontend/src/types/resume.ts
export interface ResumeAnalysis {
  score: number;
  match_score: number;

  ats_level: string;
  summary: string;

  formatting_score: number;
  keyword_score: number;
  experience_score: number;
  skills_score: number;
  education_score: number;
  readability_score: number;

  strengths: string[];
  weaknesses: string[];

  matching_skills: string[];
  missing_skills: string[];

  suggestions: string[];
}

export interface ResumeUploadResponse {
  filename: string;
  pages: number;
  analysis: ResumeAnalysis;
}

export interface CoverLetterResponse {
  cover_letter: string;
}
